### DISCUSSION TITLE
Looking forward to what opus 1.5.x can do for CPU and unstable connections

### DISCUSSION THREAD

**Discussion by 38github** [jamulussoftware/jamulus#3244] [General]
https://opus-codec.org/release/stable/2024/03/04/libopus-1_5_1.html
https://opus-codec.org/demo/opus-1.5/

**Comment by ann0see** [comment]
Thanks. Good catch. Maybe raise an issue to update opus.

**Reply by ann0see** [reply]
The update seems non trivial. Just replacing the old opus build and then fixing a linking error (add libs/opus/extensions.c) to Jamulus.pro results in a segfault if I connect to a server.

**Reply by 38github** [reply]
I wonder what might cause that. Xiph claimed it was backwards compatible. I read that the decoder at run time had to enable some parameter to be able to use some new functions.

**Comment by ann0see** [comment]
FYI, the next version will at least have the non ML based upgrades in.

**Comment by ann0see** [comment]
Adding OPUS_DRED=1to DEFINES_OPUS in Jamulus.pro now seems to allow compilation and running Jamulus.
I don't think it does much/anything at the moment.

The link says

> When building Opus, using --enable-deep-plc will compile in the deep PLC code at a cost of about 1 MB in binary size. To actually enable it at run time, you will need to set the decoder complexity to 5 or more. 

While this doesn't mention DRED, I believe we might need to bump up the decoder complexity - Not sure what this entails. Not even sure if this works for our case.
